compiler/rustc_attr/src/builtin.rs

```diff
@@ -1022,14 +1022,21 @@ pub fn find_transparency(
 
 pub fn allow_internal_unstable<'a>(
     sess: &'a Session,
-    attrs: &[Attribute],
+    attrs: &'a [Attribute],
 ) -> Option<impl Iterator<Item = Symbol> + 'a> {
-    let attr = sess.find_by_name(attrs, sym::allow_internal_unstable)?;
-    let list = attr.meta_item_list().or_else(|| {
-        sess.diagnostic()
-            .span_err(attr.span, "allow_internal_unstable expects list of feature names");
-        None
-    })?;
+    let attrs = sess.filter_by_name(attrs, sym::allow_internal_unstable);
```

Comment:
I don't see how this PR fixes anything. All you've done is replace a `?` operator with a filter map. But `find_by_name` will still only return a single attribute.

Please revert the PR. I'll reopen the issue, then we can discuss an actual fix.

Reply:
It's not find_by_name, but filter_by_name which returns an iterator with all matching attributes, instead of the first one.

Reply:
:facepalm: I apologize. I should not review half asleep

Reply:
It's fine. I'll check locally if the test case I added actually fails without